File: 2116-count-number-of-pairs-with-absolute-difference-k/2116-count-number-of-pairs-with-absolute-difference-k.c
int countKDifference(int* nums, int numsSize, int k) {
        int count = 0;
        for(int i =0;i<numsSize;i++){
            for(int j=0;j<numsSize;j++){
                if(nums[i]-nums[j]==k){
                    count++;
                }
            }
        }
        return count;
}

File: 2116-count-number-of-pairs-with-absolute-difference-k/2116-count-number-of-pairs-with-absolute-difference-k.java
class Solution {
    public int countKDifference(int[] nums, int k) {
        int count = 0;
        Map<Integer,Integer> map = new HashMap<>();
        for(int i =0;i<nums.length;i++){
            count += map.getOrDefault(nums[i]-k,0) + map.getOrDefault(nums[i]+k,0);
            map.put(nums[i],map.getOrDefault(nums[i],0)+1);
        }
        return count;
    }
}